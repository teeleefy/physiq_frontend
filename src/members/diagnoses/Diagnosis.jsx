import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
import '../styles/Diagnosis.css'


function Diagnosis({diagnosis, memberId}){

    return(
        <>
            <ListGroupItem>
            <Card className="Allergy-card">
                <CardBody>
                    <CardText>
                        <p><b>Diagnosis Name:</b> <NavLink className="Diagnosis-name" to={`/member/${memberId}/diagnoses/${diagnosis.id}`} key={diagnosis.id}>
                            {diagnosis.name}
                        </NavLink></p>
                        <p><b>Date Received:</b> {diagnosis.dateReceived}</p>
                        <p><b>Notes:</b> {diagnosis.notes}</p>
                    </CardText>
                </CardBody>
            </Card>
            </ListGroupItem>
        </>
     )
}

export default Diagnosis;