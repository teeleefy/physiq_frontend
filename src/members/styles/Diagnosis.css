.Diagnosis-card {
    height: 15vh;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Diagnosis-label {
    font-size: 1.5rem;
}

.Diagnosis-h1 {
    font-size: 3rem;
    font-weight: 600;
}

.Diagnoses-navlink {
    text-decoration: none;
    color: white;
}

.Diagnosis-date-input {
    width: 20vw;
}