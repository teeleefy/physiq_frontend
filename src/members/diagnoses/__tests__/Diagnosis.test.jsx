import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import Diagnosis from '../Diagnosis'

let diagnosis={name: "Hypertension", dateReceived: "2010/01/01", notes: "Controlled by lisinopril."}
let memberId=1000;

describe('Diagnosis', () => {
  it('renders the Diagnosis component', () => {
    renderWithRouter(<Diagnosis diagnosis={diagnosis} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Diagnosis component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Diagnosis diagnosis={diagnosis} memberId={memberId}/>);
  expect(asFragment()).toMatchSnapshot();
});