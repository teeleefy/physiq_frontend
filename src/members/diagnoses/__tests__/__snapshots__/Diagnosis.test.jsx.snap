// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Diagnosis-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Diagnosis Name:
          </b>
           
          <a
            class="Diagnosis-name"
            href="/member/1000/diagnoses/undefined"
          >
            Hypertension
          </a>
        </p>
        <p>
          <b>
            Date Received:
          </b>
           2010/01/01
        </p>
        <p>
          <b>
            Notes:
          </b>
           Controlled by lisinopril.
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
